libs/src/services/solanaWeb3Manager/transfer.js

```diff
@@ -59,61 +151,76 @@ async function transferWAudioBalance ({
   recipientSolanaAddress,
   claimableTokenPDA,
   solanaTokenProgramKey,
+  feePayerKey,
   claimableTokenProgramKey,
   connection,
-  identityService
+  mintKey,
+  transactionHandler
 }) {
-  const strippedEthAddress = senderEthAddress.replace('0x', '')
-
-  const ethAddressArr = Uint8Array.of(
-    ...new BN(strippedEthAddress, 'hex').toArray('be')
-  )
 
   const senderSolanaPubkey = new PublicKey(senderSolanaAddress)
   const recipientPubkey = new PublicKey(recipientSolanaAddress)
-  const { signature, recoveryId } = SolanaUtils.signBytes(recipientPubkey.toBytes(), senderEthPrivateKey)
 
-  const instructionData = new TransferInstructionData({
-    ethAddress: ethAddressArr,
-    amount
+  const {
+    accountNonce,
+    nonce
+  } = await getAccountNonce({
+    connection,
+    mintKey,
+    ethAddress: senderEthAddress,
+    claimableTokenProgramKey
   })
 
-  const serializedInstructionData = borsh.serialize(
-    transferInstructionSchema,
-    instructionData
-  )
-
-  const serializedInstructionEnum = Uint8Array.of(
-    1,
-    ...serializedInstructionData
-  )
-
   const accounts = [
-    // 0. `[w]` Token acc from which tokens will be send (bank account)
+    // 0. `[sw]` Fee payer
+    {
+      pubkey: feePayerKey,
+      isSigner: true,
+      isWritable: false
+    },
+    // 1. `[w]` Token acc from which tokens will be send (bank account)
     {
       pubkey: senderSolanaPubkey,
       isSigner: false,
       isWritable: true
     },
-    // 1. `[w]` Receiver token acc
+    // 2. `[w]` Receiver token acc
     {
       pubkey: recipientPubkey,
       isSigner: false,
       isWritable: true
     },
-    // 2. `[r]` Banks token account authority
+    // 3. `[w]` Nonce Account
+    {
+      pubkey: accountNonce,
+      isSigner: false,
+      isWritable: true
+    },
+    // 4. `[r]` Banks token account authority
     {
       pubkey: claimableTokenPDA,
       isSigner: false,
       isWritable: false
     },
-    // 3. `[r]` Sysvar instruction id
+    // 5. `[r]` Sysvar Rent id
+    {
+      pubkey: SYSVAR_RENT_PUBKEY,
+      isSigner: false,
+      isWritable: false
+    },
+    // 6. `[r]` Sysvar instruction id
     {
       pubkey: SYSVAR_INSTRUCTIONS_PUBKEY,
       isSigner: false,
       isWritable: false
     },
-    // 4. `[r]` SPL token account id
+    // 7. `[r]` Sysvar program id
```

Comment:
let's make this comment explicitly the 'system program id'? and maybe link to sol web3.js